dot/core/service_integration_test.go

```diff
@@ -11,31 +11,147 @@ import (
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/genesis"
 	"github.com/ChainSafe/gossamer/lib/keystore"
 	"github.com/ChainSafe/gossamer/lib/runtime"
 	runtimemocks "github.com/ChainSafe/gossamer/lib/runtime/mocks"
+	"github.com/ChainSafe/gossamer/lib/runtime/storage"
 	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
 	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
 	"github.com/ChainSafe/gossamer/lib/transaction"
 	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/lib/utils"
 	"github.com/ChainSafe/gossamer/pkg/scale"
 	"github.com/golang/mock/gomock"
+	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
 //go:generate mockgen -destination=mock_telemetry_test.go -package $GOPACKAGE github.com/ChainSafe/gossamer/dot/telemetry Client
 
+//nolint
+type testAccountData struct {
+	Free       *scale.Uint128
+	Reserved   *scale.Uint128
+	MiscFrozen *scale.Uint128
+	FreeFrozen *scale.Uint128
+}
+
+func balanceKey(t *testing.T, pub []byte) []byte {
+	var bKey []byte
+	h0, err := common.Twox128Hash([]byte("System"))
+	require.NoError(t, err)
+	bKey = append(bKey, h0...)
+	h1, err := common.Twox128Hash([]byte("Account"))
+	require.NoError(t, err)
+	bKey = append(bKey, h1...)
+	h2, err := common.Blake2b128(pub)
+	require.NoError(t, err)
+	bKey = append(bKey, h2...)
+	bKey = append(bKey, pub...)
+	return bKey
+}
+
+func newTestDigest(t *testing.T, slotNumber uint64) scale.VaryingDataTypeSlice {
+	testBabeDigest := types.NewBabeDigest()
+	err := testBabeDigest.Set(types.BabePrimaryPreDigest{
+		VRFOutput: [sr25519.VRFOutputLength]byte{
+			0, 91, 50, 25, 214, 94, 119, 36, 71, 216, 33,
+			152, 85, 184, 34, 120, 61, 161, 164, 223, 76,
+			53, 40, 246, 76, 38, 235, 204, 43, 31, 179, 28},
+		VRFProof: [sr25519.VRFProofLength]byte{
+			120, 23, 235, 159, 115, 122, 207, 206, 123, 232,
+			75, 243, 115, 255, 131, 181, 219, 241, 200, 206,
+			21, 22, 238, 16, 68, 49, 86, 99, 76, 139, 39, 0,
+			102, 106, 181, 136, 97, 141, 187, 1, 234, 183, 241,
+			28, 27, 229, 133, 8, 32, 246, 245, 206, 199, 142,
+			134, 124, 226, 217, 95, 30, 176, 246, 5, 3},
+		AuthorityIndex: 17,
+		SlotNumber:     slotNumber,
+	})
+	require.NoError(t, err)
+	data, err := scale.Marshal(testBabeDigest)
+	require.NoError(t, err)
+	vdts := types.NewDigest()
+	err = vdts.Add(
+		types.PreRuntimeDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              data,
+		},
+		types.ConsensusDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              data,
+		},
+		types.SealDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              data,
+		},
+	)
+	require.NoError(t, err)
+	return vdts
+}
+
+func generateTestValidRemarkTxns(t *testing.T, pubKey []byte, accInfo types.AccountInfo) ([]byte, runtime.Instance) {
+	projectRootPath := utils.GetProjectRootPathTest(t) + "/chain/gssmr/genesis.json"
```

Comment:
nit so it works on all weird OSes/Filesystems

```suggestion
	projectRootPath := filepath.Join(utils.GetProjectRootPathTest(t), "chain/gssmr/genesis.json")
```